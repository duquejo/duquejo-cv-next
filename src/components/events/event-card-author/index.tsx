export * from './linked-author';
export * from './static-author';
