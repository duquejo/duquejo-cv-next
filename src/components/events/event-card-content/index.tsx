export * from './push-event';
export * from './watch-event';
export * from './blog-post-event';
