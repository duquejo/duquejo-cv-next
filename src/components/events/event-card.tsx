import type { Event } from '@/interfaces';
import Link from 'next/link';
import Image from 'next/image';
import { Eye, GitBranch, Github, GitMerge, GitPullRequest } from 'lucide-react';
import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from '@/components/ui/card';
import { ReactNode } from 'react';
import { Badge } from '@/components/ui/badge';
import { cn } from '@/lib';

export const EventCard = ({ created_at, payload, actor, repo, type }: Event) => {
  const toLocaleDateString = (dateString: string) => {
    return new Date(dateString).toLocaleString();
  };

  const toIcon = (iconString: string): ReactNode => {
    const commonCss = 'inline bg-primary rounded-full transition-colors p-1';
    const commonProps = {
      size: 30,
      'data-testid': 'event-action',
    };
    switch (iconString) {
      case 'PullRequestEvent':
        return <GitPullRequest className={cn(commonCss, 'bg-yellow-400')} {...commonProps} />;
      case 'CreateEvent':
        return <GitBranch className={cn(commonCss, 'bg-teal-400')} {...commonProps} />;
      case 'PushEvent':
        return <GitMerge className={cn(commonCss, 'bg-red-400')} {...commonProps} />;
      case 'WatchEvent':
      case 'PullRequestReviewEvent':
        return <Eye className={cn(commonCss, 'bg-cyan-300')} {...commonProps} />;
      default:
        return <GitMerge className={cn(commonCss, 'bg-purple-400')} {...commonProps} />;
    }
  };

  return (
    <Card role="listitem" className="lg:max-w-xs">
      <CardHeader className="pb-3">
        <CardTitle className="flex items-center text-xs justify-between">
          <time className="text-xs font-extrabold">{toLocaleDateString(created_at)}</time>
          {payload.ref && <Badge variant="secondary">{payload.ref}</Badge>}
        </CardTitle>
        <CardDescription>
          <Link
            href={`https://github.com/${repo.name}`}
            className="flex items-center space-x-2"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <Image
              src={actor.avatar_url}
              alt={actor.display_login}
              className="max-h-7 rounded-full"
              width="28"
              height="28"
            />
            <span className="font-semibold underline underline-offset-2 decoration-1 truncate text-xs">
              {repo.name}
            </span>
          </Link>
        </CardDescription>
      </CardHeader>
      <CardContent className="text-xs leading-relaxed text-left break-words">
        {payload.description && <strong className="font-bold">{payload.description}</strong>}
        {payload.commits &&
          payload.commits.length > 0 &&
          payload.commits.map(
            (commit) =>
              !['dependabot[bot]'].includes(commit.author.name) && (
                <div key={commit.sha}>
                  <p className="inline text-muted-foreground">{commit.message}</p>
                  {type === 'PushEvent' && (
                    <Link
                      href={`${commit.url.replace('https://api.github.com/repos', 'https://github.com')}`}
                      className="inline font-semibold"
                      target="_blank"
                      rel="noopener noreferrer nofollow"
                      title="Link"
                    >
                      <Github className="ml-2 inline bg-primary rounded-full p-1 h-5 w-5 text-background transition-colors hover:bg-primary-foreground hover:text-foreground" />
                    </Link>
                  )}
                </div>
              ),
          )}
      </CardContent>
      <CardFooter className="flex justify-between">
        <span className="text-background">{toIcon(type)}</span>
        <span className="text-muted-foreground">{type}</span>
        {payload.ref_type && <span className="capitalize">({payload.ref_type})</span>}
      </CardFooter>
    </Card>
  );
};
