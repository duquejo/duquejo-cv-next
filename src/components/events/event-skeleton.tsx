import { Skeleton } from '@/components/ui/skeleton';

interface Props {
  rounds?: number;
}

export const EventSkeleton = ({ rounds = 2 }: Props) => {
  return new Array(rounds).fill(0).map((_, i) => (
    <div key={i} className="flex flex-col space-y-3">
      <div className="flex items-center justify-between gap-x-3">
        <Skeleton className="flex-1 h-8 w-full" />
        <Skeleton className="h-8 w-[100px]" />
      </div>
      <Skeleton className="h-[125px] w-full" />
      <div className="space-y-3">
        <Skeleton className="h-4 w-full" />
        <Skeleton className="h-4 w-full" />
      </div>
      <div className="flex items-center space-x-4">
        <Skeleton className="h-12 w-12 rounded-full" />
        <div className="space-y-3">
          <Skeleton className="h-4 w-[250px]" />
          <Skeleton className="h-4 w-[200px]" />
        </div>
      </div>
    </div>
  ));
};
