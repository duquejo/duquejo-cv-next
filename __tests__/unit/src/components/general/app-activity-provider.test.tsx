import { render, screen } from '@testing-library/react';
import { EventProvider } from '@/components/events/event-provider';
import { SWRConfig } from 'swr';

vi.mock('next-intl', () => ({
  useTranslations: () => (key: string) =>
    ({
      title: 'title',
      description: 'description',
    })[key],
}));

describe('<EventProvider /> tests', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <EventProvider>
          <div data-testid="child">Child Component</div>
        </EventProvider>
      </SWRConfig>,
    );

    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('child')).toBeInTheDocument();
  });

  it('should toggle loadEvents state on sheet open', async () => {
    const { container } = render(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <EventProvider defaultOpen={true}>
          <div data-testid="child">Child Component</div>
        </EventProvider>
      </SWRConfig>,
    );

    expect(container).toMatchSnapshot();
  });
});
