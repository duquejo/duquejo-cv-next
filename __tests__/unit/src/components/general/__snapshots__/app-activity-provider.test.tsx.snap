// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AppActivityProvider /> tests > should match the snapshot 1`] = `
<div>
  <div
    data-testid="child"
  >
    Child Component
  </div>
</div>
`;

exports[`<AppActivityProvider /> tests > should toggle loadEvents state on sheet open 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
>
  <div
    data-testid="child"
  >
    Child Component
  </div>
</div>
`;
